packages/cli/src/services/ownership.service.ts

```diff
@@ -50,4 +51,25 @@ export class OwnershipService {
 			ownedBy: ownerId ? { id: ownerId } : null,
 		});
 	}
+
+	addOwnedByAndSharedWith(_credential: CredentialsEntity): Credentials.WithOwnedByAndSharedWith {
```

Comment:
Also can we add some unit tests covering this function?

Reply:
> Simply because we might do the same for workflows, right? We have the addOwnedBy above that could be renamed to contain also clarification about what it applies to (maybe out of context of this PR)

I think this is clear from the typing? Different entities require different fields, and the caller cannot misuse the methods - the typing will block them if they try to.

> Also can we add some unit tests covering this function?

Will do!